cni/pkg/util/podutil.go

```diff
@@ -56,6 +56,11 @@ func PodRedirectionEnabled(namespace *corev1.Namespace, pod *corev1.Pod) bool {
 		// Pod explicitly asked to not have ambient redirection enabled
 		return false
 	}
+	if pod.Spec.HostNetwork {
```

Comment:
Yeah we already have this check effectively in place for sidecar injection too and have for some time, shoulda been here.